### PR TITLE
dbus-broker: 29 -> 31, use external c-util dependencies

### PR DESCRIPTION
**Pull request by xaverdh**
###### Description of changes

Update to v31 (https://github.com/bus1/dbus-broker/blob/main/NEWS.md).

Different approach to https://github.com/NixOS/nixpkgs/pull/172845, we now unvendor all `c-util` dependencies.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by xaverdh** [comment]
Ah I think the issue is that contrary to what is stated in the [c-stdaux source](https://github.com/c-util/c-stdaux/blob/7a8493bebc595f94ea57fa1cb6a765a66185aa95/meson.build#L22), the `cflags` variable is *not* exported and does not end up in the generated `.pc`-file.

**Comment by xaverdh** [comment]
This is ready for review @SuperSandro2000 @peterhoeg

**Comment by xaverdh** [comment]
Bumped version to 31, which was just released. Also added myself as maintainer of `dbus-broker`.

**Comment by xaverdh** [comment]
note: the `pkg-config` issue was resolved upstream

**Comment by xaverdh** [comment]
rebased

**Comment by xaverdh** [comment]
@SuperSandro2000 could you take a look?

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -374,6 +374,20 @@ with pkgs;
...
+
+  c-utf8 = callPackage ../development/libraries/c-util/c-utf8 { };
+
+  c-rbtree = callPackage ../development/libraries/c-util/c-rbtree { };
+
+  c-shquote = callPackage ../development/libraries/c-util/c-shquote { };
+
+  c-ini = callPackage ../development/libraries/c-util/c-ini { };
+
+  c-dvar = callPackage ../development/libraries/c-util/c-dvar { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:389]: This should besorted alphabetical.

File: pkgs/os-specific/linux/dbus-broker/default.nix [hunk 2]
```diff
@@ -1,19 +1,22 @@
...
+    owner = "bus1";
+    repo = "dbus-broker";
+    rev = "v${version}";
+    sha256 = "1v904wcf4wsysagbg02177slwaqh1ps1qqa2ayab8ljxdv0bbnvx";
   };
 
-  nativeBuildInputs = [ docutils meson ninja pkg-config ];
+  nativeBuildInputs = [
+    docutils meson ninja pkg-config
+    c-stdaux c-list c-utf8 c-rbtree c-shquote c-ini c-dvar
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/dbus-broker/default.nix:18]: You are sure this are not buildInputs?

File: pkgs/development/libraries/c-util/c-dvar/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
+{ lib, stdenv, fetchFromGitHub, pkg-config, meson, ninja
+, c-stdaux, c-utf8 }:
+
+stdenv.mkDerivation rec {
+  pname = "c-dvar";
+  version = "unstable-2022-05-03";
+
+  nativeBuildInputs = [
+    meson ninja pkg-config
+    c-stdaux c-utf8
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/c-util/c-dvar/default.nix:10]: This line is buildInputs.

**Review by SuperSandro2000 — COMMENTED** [review]
The websites are not very informative at all. We should just link the repository instead.

Please copy the short description from the github repositories instead of keeping one generic one.

nativeBuildInputs gets usually sorted after src, to refleect the order they are used in stdenv.

File: pkgs/os-specific/linux/dbus-broker/default.nix (hunk 2 above)
Comment (peterhoeg) [reply pkgs/os-specific/linux/dbus-broker/default.nix:18]: They are definitely buildInputs. I'm so sorry, I didn't see this PR before I made #185562. Do you want to update this PR or should we just go ahead and use mine?

**Comment by xaverdh** [comment]
closing this since https://github.com/NixOS/nixpkgs/pull/185562 landed

**Pull request closed without merging**